internal/vfs/vfstest/vfstest.go

```diff
@@ -168,6 +170,28 @@ func (m *mapFS) open(p canonicalPath) (fs.File, error) {
 	return m.m.Open(string(p))
 }
 
+func (m *mapFS) remove(path string) error {
+	canonical := m.getCanonicalPath(path)
+	canonicalString := string(canonical)
+	fileInfo, err := m.m.Stat(canonicalString)
+	if err != nil {
+		// file not found
+		return nil
+	}
+	delete(m.m, canonicalString)
+	delete(m.symlinks, canonical)
+
+	if fileInfo.IsDir() {
+		for path := range m.m {
+			if strings.HasPrefix(path, canonicalString) {
```

Comment:
This needs to be HasPrefix on `canonicalString + "/"`, otherwise removing `/foo/bar` would remove `/foo/barbar`.